impeller/fixtures/planet.frag

```diff
@@ -0,0 +1,486 @@
+    // (to-do)
+    vec3 worldspace_surface_normal =
+        mix(worldspace_normal, calcNormal(noise_coord, worldspace_normal),
+            water_selector);
+
+    if (frag_info.show_normals > 0.0) {
+      planet_color = worldspace_surface_normal;
+    }
+    float wrap = 0.05;
+    // float dp = max(0.0, dot(worldspace_light_direction,
+    // worldspace_surface_normal));  // dot product nvida surface scattering
+    // trick
+    float dp = max(
+        0.0,
+        (dot(worldspace_light_direction, worldspace_surface_normal) + wrap) /
+            (1.0 + wrap));
+
+    vec3 darkRed = vec3(0.25, 0.0, 0.0);
+    vec3 lightColor = mix(darkRed, vec3(0.75), smoothstep(0.05, 0.5, dp));
+
+    vec3 ambient = vec3(0.002);  // lets not have complete darkness
+    vec3 diffuse = lightColor * dp;
+
+    vec3 r = normalize(
+        reflect(-worldspace_light_direction, worldspace_surface_normal));
+    float phongValue = max(0.0, dot(wsViewDir, r));
+    phongValue = pow(phongValue, spec_params.y);
+
+    vec3 specular = vec3(phongValue) * spec_params.x * diffuse;
+
+    vec3 planetShading = planet_color * (diffuse + ambient) + specular;
+    planet_color = planetShading;
+
+    // Fresnel for atmosphere
+    float fresnel = smoothstep(1.0, 0.1, view_normal.z);  // z == from camera.
+    // "blue halo goes around the dark side of the planet
+    // fresnel = pow(fresnel, 8.0);
+    fresnel = pow(fresnel, 8.0) * dp;
+    planet_color = mix(planet_color, vec3(0.0, 0.5, 1.0), fresnel);
+
+    if (frag_info.show_noise > 0.0) {
+      planet_color = vec3(noise_sample);
+    }
+  }
+  // (color -> planet_color) when "d" is between 0 and -1 (inside the circle)
+  color = mix(color, planet_color, smoothstep(0.0, -1.0, d));
+
+  // Atmospheric glow
+  // 40 pixels outside the planet
+  if (d < atmosphere_thickness && d >= -1.0) {
+    // color = vec3(1.0); // draw the halo
+    float planetSizeAtmos = planet_size + atmosphere_thickness;
+    mat3 planet_rotation = rotateY(frag_info.time * rotation_speed);
+
+    float x = pixel_coords.x / planetSizeAtmos;
+    float y = pixel_coords.y / planetSizeAtmos;
+    float z = sqrt(1.0 - x * x - y * y);
+    vec3 normal = planet_rotation * vec3(x, y, z);
+
+    float lighting = dot(normal, normalize(vec3(0.5, 1.0, 0.5)));
+    lighting =
+        smoothstep(-0.15, 1.0, lighting);  // same as above; just no wrap.
+
+    vec3 glow_color =
+        vec3(0.05, 0.3, 0.9) * exp(-0.01 * d * d) * lighting * 0.75;
+    color += glow_color;
+  }
+
+  return color;
+}
+
+void main() {
+  vec2 vUvs = (gl_FragCoord.xy - 0.5) / frag_info.resolution;
+  vUvs.y = 1.0 - vUvs.y;  // flip flutter's upside down.
+  vec2 pixel_coords = (vUvs - 0.5) * frag_info.resolution;
+
+  vec3 color = vec3(0.0);
+  color = GenerateStars(pixel_coords);
+  color = DrawPlanet(pixel_coords, color, frag_info.planet_size,
+                     frag_info.planet_size * 0.1, frag_info.speed);
+
+  frag_color = vec4(pow(color, vec3(1.0 / 2.2)), 1.0);
+}
```

Comment:
Needs newline

Reply:
Is this something `et format` needs to handle better?

Reply:
Yeah not sure why clang-format isn't doing this anymore. Filed a quick issue: https://github.com/flutter/flutter/issues/150148